### PR TITLE
fix: remove version patches and add MM webhooks

### PR DESCRIPTION
**Pull request by siggiskulason**
device-camera-go and device-rest-go do not need to be patched as
the upstream repositories have been fixed.
Also added MM messages on sync and LP failures.

### CODE REVIEW

File: .github/workflows/wf-device-camera.yml [hunk 1]
```diff
@@ -6,13 +6,16 @@ jobs:
...
     runs-on: ubuntu-latest
     steps:      
-      - name: Build and version device-camera
-        uses: canonical/edgex-sync-and-create-launchpad-branch-action@v1.4.0
-        with:
-          edgex-repo: "device-camera-go"
-          canonical-repo: "edgex-device-camera-go"
-          patch-file: "0001-snapcraft-version-patch.diff"
-          ssh-private-key: ${{ secrets.SSH_CANONICAL_EDGEX_DEVICE_CAMERA }}
+    - name: Build and version device-camera
```
Comment (tonyespy) [inline .github/workflows/wf-device-camera.yml]: Should version still be included in the name?

File: .github/workflows/wf-device-camera.yml [hunk 2]
```diff
@@ -43,10 +46,14 @@ jobs:
...
     needs:  build-snap
     steps:
-      - name: Kick off Launchpad build
-        uses: canonical/edgex-launchpad-build-action@v1.4
-        with:
-          edgex_snap: "edgex-device-camera"
-          consumer_name: ${{ secrets.LP_CONSUMER_NAME }}
-          access_token: ${{ secrets.LP_ACCESS_TOKEN }}
-          access_secret: ${{ secrets.LP_ACCESS_SECRET }}
+    - name: Kick off Launchpad build
```
Comment (tonyespy) [inline .github/workflows/wf-device-camera.yml]: Looks like the formatting was updated too, was this intentional?

File: .github/workflows/wf-device-rest.yml [hunk 3]
```diff
@@ -6,13 +6,16 @@ jobs:
...
     runs-on: ubuntu-latest
     steps:
-      - name: Sync and version repo
-        uses: canonical/edgex-sync-and-create-launchpad-branch-action@v1.4.0
-        with:
-          edgex-repo: "device-rest-go"
-          canonical-repo: "edgex-device-rest-go"
-          ssh-private-key: ${{ secrets.SSH_CANONICAL_EDGEX_DEVICE_REST }}  
-          patch-file: "0001-snapcraft-version-patch.diff" 
+    - name: Sync and version repo
```
Comment (tonyespy) [inline .github/workflows/wf-device-rest.yml]: Same as previous comment...

File: .github/workflows/wf-device-rest.yml [hunk 4]
```diff
@@ -43,10 +46,14 @@ jobs:
...
     needs:  build-snap
     steps:
-      - name: Kick off Launchpad build
-        uses: canonical/edgex-launchpad-build-action@v1.4
-        with:
-          edgex_snap: "edgex-device-rest"
-          consumer_name: ${{ secrets.LP_CONSUMER_NAME }}
-          access_token: ${{ secrets.LP_ACCESS_TOKEN }}
-          access_secret: ${{ secrets.LP_ACCESS_SECRET }}         
+    - name: Kick off Launchpad build
```
Comment (tonyespy) [inline .github/workflows/wf-device-rest.yml]: formatting?

**Review by tonyespy — COMMENTED** [review]
A couple of comments inline...

Also, I think it probably makes sense to split into two commits, one for camera, and one for rest.

**Comment by tonyespy** [comment]
Also, two other comments:

 - since commits should be logical changes you might also want to break out the change for each which drops the version patch and the change that adds the MM alert
 - and finally, in the commit summary, maybe change "patch" to "version patch"?

File: .github/workflows/wf-device-camera.yml (hunk 1 above)
Comment (siggiskulason) [reply .github/workflows/wf-device-camera.yml]: In this case, "version" refers to the process of creating the VERSION file, which is still done there - it's just the patching of the versioning logic which has been removed.

File: .github/workflows/wf-device-camera.yml (hunk 2 above)
Comment (siggiskulason) [reply .github/workflows/wf-device-camera.yml]: It was - It wasn't uniform across the three steps so I cleaned that up. Perhaps that should have been left out of this commit though as it's not related to the change which is actually being made.

**Comment by siggiskulason** [comment]
Thanks @tonyespy  - Yes, this PR did sort of break the cardinal rule of making distinct, logical changes. It is really making three changes
1) minor changes to tabs to make the indentation standard across the steps
2) changes to remove the reference to a patch file - and to delete that patch file, for both device-camera and device-rest
3) adding the MM webhook.

I'll break this up into separate commits
1) removing device-camera-go patch reference / and delete that patch file
2) removing device-rest-go patch reference / and delete that patch file
3) add webhooks in device-camera
4) add webhooks in device-rest

**Review by tonyespy — APPROVED** [review]
LGTM

**Pull request merged**